manifests/init.pp

```diff
@@ -55,28 +56,39 @@
       $prefer_open_vm_tools_real = $prefer_open_vm_tools
     }
 
-    # OSs that have open-vm-tools
-    case $::operatingsystem {
-      'RedHat', 'CentOS': {
-        $_use_open_vm_tools = $lsbmajdistrelease_int >= 7
-      }
-      'SLED', 'SLES': {
-        $_use_open_vm_tools = $lsbmajdistrelease_int >= 12
-      }
-      'OpenSuSE': {
-        $_use_open_vm_tools = $lsbmajdistrelease_int >= 12
-      }
-      'Ubuntu': {
-        if $prefer_open_vm_tools_real == true {
-          # include Ubuntu 12.04
+    if is_string($force_open_vm_tools) == true {
```

Comment:
Yeah ! Take care for stringified variable types. For testing I recommend to use the newest incarnation of my variable type validations https://github.com/ghoneycutt/puppet-module-ssh/blob/v3.34.0/spec/classes/init_spec.rb#L3482-L3511